### PR TITLE
fix(cli): improve repo error/help UX

### LINKED ISSUE #399: Improve CLI error/help UX: friendly missing-WORKFLOW.md error, --json on error paths, surfaced issue arg format, and unknown-subcommand handling
## Problem
Several small CLI rough edges share a root cause — error/help paths that don't match the project's own conventions:
- **Raw ENOENT.** `repo init` with no WORKFLOW.md prints a bare `ENOENT: no such file or directory, open '.../WORKFLOW.md'` (exit 1) instead of pointing to `workflow init`. `packages/cli/src/repo-runtime.ts:74-75` reads with no existence check; the raw error propagates via `packages/cli/src/commands/repo.ts:103-108`. `doctor.ts:642-643` already has the friendly wording.
- **--json ignored on error.** `--json` is global (`index.ts:86-92`) but error paths for `repo init/status/run` write plain text to stderr even with `--json` (e.g. `repo status --json` -> 'No repository runtime config found...'), so scripts can't parse failures (repo.ts:103-108, status.ts:238-241/313-316, project-selection.ts:185-194).
- **Issue arg format hidden.** `repo run/explain --help` shows `[args...]` ('Issue identifier and passthrough options'); the required `owner/repo#number` surfaces only on error (`index.ts:562,616`; run.ts:69). The project's own `workflow` commands use the better `--issue <owner/repo#number>` form (index.ts:293,348).
- **Confusing unknown-subcommand error.** `repo badcmd` / `workflow badcmd` print `too many arguments for 'repo'` instead of `unknown command` and don't point to `--help`, because the parent commands take no positional args (index.ts:424-431, 232).
- **Removed subcommands shown.** `repo --help` lists `list/add/remove/sync` as 'Removed' (index.ts:433-479, no `{hidden:true}`) while top-level removed commands are hidden.

## Impact (OSS user)
New users hit cryptic errors with no next step, scripts break on unparseable failures, and `--help` is vaguer than the root help — friction during the first attempts.

## Evidence (real files)
See file:line refs above; all reproduced against the built dist.

## Proposed fix
Add an existence check + friendly error in `initRepoRuntime` mirroring doctor. Emit `{error:{code,message}}` to stdout on error paths for scriptable repo commands when `--json` is set. Change `repo run/explain` argument to `<issue>` ('owner/repo#number') with an example. Add unknown-command handling on the `repo`/`workflow` parents pointing to `--help`. Register the removed `repo` subcommands with `{hidden:true}`.

## Re-examination (re-verified against current code)
All five sub-claims verified against current code AND reproduced against the built dist (packages/cli/dist/index.js, v0.4.5). (1) CONFIRMED: repo-runtime.ts:75 does `parseWorkflowMarkdown(await readFile(workflowPath,"utf8"))` with no existence check; `repo init` in an empty dir prints raw `ENOENT: no such file or directory, open '.../WORKFLOW.md'` exit 1, while doctor.ts:642-643 already has the friendly 'Run gh-symphony workflow init' wording. (2) CONFIRMED: `--json` is a global option (index.ts:91) honored on success (repo.ts:92) but ignored on all error paths — `repo init --json` and `repo status --json` both emit plain text to stderr (repo.ts:104-107, status.ts:314-315, project-selection.ts:190-193 handleMissingManagedProjectConfig which takes no args so cannot see --json). Note the structured kind/message data already exists internally at project-selection.ts:62-76. (3) CONFIRMED: repo run/explain use `.argument('[args...]','Issue identifier and passthrough options')` (index.ts:562,616); `repo run --help` shows vague `args`, while workflow preview/doctor use `--issue <owner/repo#number>` (index.ts:293,348). Added nuance: the curated top-level help (help.ts:84-96 + snapshot) ALREADY advertises `repo run <issue>`/`repo explain <issue>`, so the subcommand-level help diverges from the project's own top-level help. (4) PARTIALLY CONFIRMED with correction: `repo badcmd`/`workflow badcmd` do print the confusing `error: too many arguments for 'repo'. Expected 0 arguments but got 1.` (parents at index.ts:424-431,232 take no positional args) — BUT they DO already print `(run with --help for usage)`, contradicting the candidate's claim that they 'don't point to --help'. (5) CONFIRMED: repo subcommands list/add/remove/sync (index.ts:433-479) are registered without `{hidden:true}` and appear in `repo --help`, whereas top-level removed commands use registerRemovedCommand with `{hidden:true}` (index.ts:193).

**Deeper context / scenario:** Root cause: error/help code was written piecemeal per-command rather than against a shared convention, so the polished patterns the project already established elsewhere (doctor's friendly remediation text, the global --json contract, the curated top-level help's `<issue>` form, and the `{hidden:true}` treatment for removed commands) were never propagated to the repo subcommands. Blast radius is the entire first-run surface of the npm-distributed CLI (@gh-symphony/cli bin gh-symphony): repo init/status/run/explain plus the repo and workflow help. Concrete scenario: I install the CLI, clone a repo, and run `gh-symphony repo init` before authoring WORKFLOW.md — I get a bare `ENOENT ... open '/path/WORKFLOW.md'` with no hint that I should run `gh-symphony workflow init`, even though doctor would have told me exactly that. Later I script `repo status --json` into CI expecting machine-readable output, but on the (common) no-config path it prints unparseable prose to stderr, so my pipeline silently mis-handles the failure. When I look for the issue syntax via `repo run --help`, it just says `args`, so I guess wrong and only learn `owner/repo#number` from a usage error. Each item is individually minor (true P2) but they cluster at the first-15-minutes funnel where they compound. Nuance the original missed: the unknown-subcommand error already points to --help (so that half-claim is overstated), and the `<issue>` rename is essentially a consistency fix since help.ts already uses that exact label — meaning the fix mostly aligns code with the project's own existing copy, and the structured error data for the --json path already exists in resolveProjectSelection.

---
_Priority: P2 (re-ranked **P2-A**, overall #4 of 16) · impact: medium · reach: many-users · type: ux-polish · effort: M_
_Re-classified from an OSS-user-perspective P2 re-assessment._

### PR DESCRIPTION
**Pull request by moncher-dev**
## TL;DR

- Fixes the repo CLI first-run/error UX from #399.
- `repo init` now reports a friendly missing `WORKFLOW.md` remediation, repo error paths honor `--json`, and repo run/explain help surfaces the `<issue>` format.
- Rework keeps missing-issue validation inside command handlers so `--json` remains structured even when the issue is omitted, and unknown-subcommand routing now tolerates global option placement.

## 변경 지점 다이어그램

- `packages/cli/src/repo-runtime.ts` → detects missing workflow files before parsing.
- `packages/cli/src/cli-error.ts` → centralizes text vs JSON error output.
- `packages/cli/src/commands/{repo,status,run,repo-explain}.ts` → uses structured error output on failure paths.
- `packages/cli/src/index.ts` → improves repo/workflow subcommand help and unknown-subcommand handling; derives namespace subcommands from Commander registration.
- `packages/cli/src/**/*.test.ts` → regression coverage for friendly errors, JSON errors, help text, hidden removed commands, unknown subcommands, and rework edge cases.

## 여기부터 보세요

- `packages/cli/src/cli-error.ts`
- `packages/cli/src/repo-runtime.ts`
- `packages/cli/src/index.ts`
- `packages/cli/src/commands/repo.test.ts`
- `packages/cli/src/index.test.ts`

## 위험 & 롤백

- Risk: low; changes are limited to CLI error/help paths and command metadata.
- Rollback: revert this PR to restore the previous CLI behavior.

## 변경 파일

- `.changeset/friendly-repo-cli-errors.md`
- `packages/cli/src/cli-error.ts`
- `packages/cli/src/repo-runtime.ts`
- `packages/cli/src/project-selection.ts`
- `packages/cli/src/commands/repo.ts`
- `packages/cli/src/commands/status.ts`
- `packages/cli/src/commands/run.ts`
- `packages/cli/src/commands/repo-explain.ts`
- `packages/cli/src/index.ts`
- CLI regression tests

## Evidence

- `pnpm --filter @gh-symphony/cli test -- src/index.test.ts src/commands/lifecycle.test.ts src/commands/repo-explain.test.ts`
- `pnpm lint`
- `pnpm test`
- `pnpm typecheck`
- `pnpm build`
- Built CLI smoke checks for `repo run --json --watch`, `repo explain --json`, `--json repo badcmd`, `repo --json badcmd`, `repo badcmd --json`, `toString foo`, `repo run --help`, and `repo explain --help`.

## Issues — Closed #399

Closes #399

## 머지 후/사람 확인

- [ ] Optional manual CLI sanity check from a clean install.

### CODE REVIEW

File: packages/cli/src/commands/run.ts [hunk 1]
```diff
@@ -60,23 +61,33 @@ const handler = async (
...
+    writeCliError({
+      code: "invalid_arguments",
+      message: parsed.error,
+      json: options.json,
+      exitCode: 2,
+    });
     return;
   }
 
   if (!parsed.issue) {
```
Comment (hojinzs) [inline packages/cli/src/commands/run.ts:73]: Nit (non-blocking, dead code): now that `repo run`'s argument is the required `<issue>` (index.ts), commander rejects a missing issue with `error: missing required argument 'issue'` **before** this handler runs — I confirmed this against the built CLI. So this `if (!parsed.issue)` branch (and its friendly `Usage: gh-symphony repo run <owner/repo#number>` text + exit code 2) is now unreachable via the CLI, and the format the issue wanted to *surface* is shown only in `--help`, not on the error path. Reachable only by direct handler unit tests. Either drop the dead branch or, if you want the friendly format on the bare-`repo run` error path, keep `<issue>` optional and let this branch own the message. Minor — help text already covers the format, so not worth blocking.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/index.ts [hunk 2]
```diff
@@ -86,6 +87,24 @@ const COMMANDS: Record<LoaderKey, () => Promise<{ default: CommandHandler }>> =
     version: () => import("./commands/version.js"),
   };
 
+const NAMESPACE_COMMANDS: Record<string, readonly string[]> = {
```
Comment (hojinzs) [inline packages/cli/src/index.ts]: Nit (non-blocking, maintainability): `NAMESPACE_COMMANDS` hardcodes the repo/workflow subcommand names, so it must be kept in sync by hand whenever a subcommand is added/renamed. If it drifts, a real command gets falsely reported as `unknown command`. Consider deriving the list from the registered `Command` children (e.g. `repo.commands.map(c => c.name())`) instead of a literal, so the source of truth stays single. Fine to defer.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — APPROVED** [review]
## ✅ Approve — #399 모든 항목 반영 확인

**요구사항 대비 검증 (issue #399 5개 항목):**
| 항목 | 상태 | Smoke 결과 (built CLI) |
|---|---|---|
| (1) `repo init` raw ENOENT → friendly remediation | ✅ | `WORKFLOW.md was not found at … Run 'gh-symphony workflow init' …` (exit 1, "ENOENT" 미노출) |
| (2) error 경로에서 `--json` 존중 | ✅ | `repo init --json` → `{"error":{"code":"missing_workflow_file","message":…}}` (stdout, exit 1) |
| (3) `repo run/explain --help` 에 `owner/repo#number` 형식 노출 | ✅ | `Usage: … run [options] <issue>` + `Issue identifier (owner/repo#number)` + Examples |
| (4) unknown subcommand 처리 | ✅ | `repo badcmd` / `workflow badcmd` → `error: unknown command 'badcmd' for 'repo'` + `(run with --help for usage)` |
| (5) 제거된 repo 서브커맨드 숨김 | ✅ | `repo --help` 에 list/add/remove/sync 미노출, 호출 시 Removed 메시지 유지 |

**Build / Test:** `pnpm --filter @gh-symphony/cli build` 성공, 전체 466 tests 통과. 신규 회귀 테스트(friendly error, JSON error, help 형식, hidden 커맨드, unknown subcommand)가 각 항목을 커버합니다. 변경 파일 커버리지도 `cli-error.ts` 100% 등 양호.

**코드 품질:** `writeCliError` 로 text/JSON 출력을 한 곳에 모은 설계가 깔끔하고, 프로젝트 기존 컨벤션(doctor remediation 문구, 전역 `--json` 계약, `{hidden:true}`)과 잘 정렬됩니다.

인라인으로 비차단 nit 2건(run.ts 의 unreachable 분기, `NAMESPACE_COMMANDS` 수동 동기화)을 남겼습니다 — 둘 다 동작에는 영향 없어 merge 차단 사유 아님. LGTM 🎉

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/index.ts [hunk 3]
```diff
@@ -180,6 +199,32 @@ function hasVersionFlag(argv: string[]): boolean {
   return argv.some((arg) => arg === "--version" || arg === "-V");
 }
 
+function handleUnknownNamespaceCommand(argv: string[]): boolean {
+  const namespace = argv[0];
+  const subcommand = argv[1];
```
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/cli/src/index.ts]: Resolved in `9da5912`: unknown-subcommand routing now extracts namespace/subcommand after skipping global options (`--json`, `--config`, `--config-dir`, `--no-color`, `--verbose`). It also uses the registered Commander namespace command tree for the valid subcommand list.

Covered by tests and built CLI smoke checks for `--json repo badcmd`, `repo --json badcmd`, and `repo badcmd --json`.

File: packages/cli/src/index.ts [hunk 4]
```diff
@@ -580,15 +634,19 @@ function createProgram(): { program: Command; wasInvoked: () => boolean } {
     repo
       .command("run")
       .description("Dispatch a single issue from the current repository")
-      .argument("[args...]", "Issue identifier and passthrough options")
+      .argument("<issue>", "Issue identifier (owner/repo#number)")
```
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/cli/src/index.ts]: Resolved in `9da5912`: `repo run` and `repo explain` keep the Commander argument optional internally while overriding `usage()` so help still displays `<issue>`. Missing issues now reach the command handlers and preserve `{error:{code,message}}` JSON output for `--json`.

Covered by `src/index.test.ts` cases for `repo run --json --watch` and `repo explain --json`, plus built CLI smoke checks.

File: packages/cli/src/commands/run.ts (hunk 1 above)
Comment (moncher-dev) [reply packages/cli/src/commands/run.ts:73]: Resolved in `9da5912`: the missing-issue branch is reachable again because Commander no longer rejects the invocation before `commands/run.ts` executes. Help still shows `Usage: gh-symphony repo run [options] <issue>` via an explicit `usage()` override.

Covered by `repo run --json --watch` entrypoint coverage and a built CLI smoke check.

File: packages/cli/src/index.ts [hunk 5]
```diff
@@ -180,6 +199,32 @@ function hasVersionFlag(argv: string[]): boolean {
...
+  const namespace = argv[0];
+  const subcommand = argv[1];
+  if (
+    !namespace ||
+    !subcommand ||
+    subcommand.startsWith("-") ||
+    !(namespace in NAMESPACE_COMMANDS)
+  ) {
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/cli/src/index.ts]: Resolved in `9da5912`: the manual object lookup is gone. The handler now finds the namespace from `program.commands` and only handles the explicit `repo`/`workflow` namespaces, so inherited property names are not treated as namespace command tables.

Covered by a regression test and built CLI smoke check for `toString foo`.

File: packages/cli/src/index.ts (hunk 2 above)
Comment (moncher-dev) [reply packages/cli/src/index.ts]: Resolved in `9da5912`: `NAMESPACE_COMMANDS` was removed. The unknown-subcommand handler now derives valid child names from the registered Commander namespace command, so hidden removed commands and future registered commands share one source of truth.

File: packages/cli/src/index.ts [hunk 6]
```diff
@@ -180,6 +181,79 @@ function hasVersionFlag(argv: string[]): boolean {
...
+  const positionals: string[] = [];
+  for (let i = 0; i < argv.length; i += 1) {
+    const arg = argv[i];
+    if (!arg) {
+      continue;
+    }
+    if (arg === "--config" || arg === "--config-dir") {
+      i += 1;
+      continue;
+    }
```
Comment (hojinzs) [inline packages/cli/src/index.ts:196]: Nit (non-blocking, latent maintainability): this skip-list only consumes the *value* of `--config`/`--config-dir`. For any *future* value-taking global option, the flag name is skipped by the generic `arg.startsWith("-")` branch below, but its value is not — so `gh-symphony repo --future-opt someval badcmd` would read `someval` as the subcommand and mis-report `unknown command 'someval'`. This is the same "manual sync" smell already flagged for `NAMESPACE_COMMANDS` (which you nicely resolved by deriving from the Commander tree). Today only `--config`/`--config-dir` take values, so this is correct as-is — just worth a comment, or deriving the value-taking option names from `program.options`, so it can't drift. Not worth blocking merge.

_Generated by [Claude Code](https://claude.ai/code)_

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — APPROVED** [review]
## ✅ Approve — 완성도 검증 완료 (#399)

PR #419 의 작업 내용을 원본 이슈 #399 의 수용 조건 대비 독립 검증했습니다. 빌드된 CLI(`packages/cli/dist/index.js`) 기준 블랙박스 스모크 테스트로 5개 요구 항목과 봇 리뷰 지적 3건의 해소 여부를 모두 재현 확인했습니다.

### 요구사항 대비 검증 (issue #399 — 5개 항목)
| 항목 | 상태 | Smoke 결과 (built CLI) |
|---|---|---|
| (1) `repo init` raw ENOENT → 친절한 remediation | ✅ | `WORKFLOW.md was not found at … Run 'gh-symphony workflow init' …` (exit 1, "ENOENT" 미노출) |
| (2) error 경로에서 `--json` 존중 | ✅ | `repo init --json`, `repo status --json` → `{"error":{"code":…,"message":…}}` (stdout, stderr 무출력) |
| (3) `repo run/explain --help` 에 `<issue>` 형식 노출 | ✅ | `Usage: … run [options] <issue>` + `Issue identifier (owner/repo#number)` + Examples |
| (4) unknown subcommand 처리 | ✅ | `repo badcmd` / `workflow badcmd` → `error: unknown command 'badcmd' for 'repo'` + `(run with --help for usage)` |
| (5) 제거된 repo 서브커맨드 숨김 | ✅ | `repo --help` 에 list/add/remove/sync 미노출, 직접 호출 시 Removed 메시지 + exit 2 유지 |

### 봇 리뷰 지적 해소 재확인 (claim이 아닌 실제 재현)
- **Codex #1** (missing-issue 시 `--json` 보존): `repo run --json --watch`, `repo explain --json` 모두 `{"error":{"code":"invalid_arguments",…}}` JSON 출력 (Commander 텍스트 아님) ✅
- **Codex #2** (글로벌 옵션 배치): `--json repo badcmd`, `repo --json badcmd`, `repo badcmd --json` 세 배치 모두 동일한 `unknown_command` JSON 출력 ✅
- **Copilot** (상속 프로퍼티 크래시): `toString foo` → 크래시 없이 `unknown command 'toString'` ✅

### Smoke / Test / Build
- `pnpm --filter @gh-symphony/cli test` → **472 passed**
- `pnpm --filter @gh-symphony/cli build` → success
- 위 모든 항목을 빌드 산출물 기준 CLI 실행으로 직접 재현

### 코드 품질
`writeCliError` 로 text/JSON 출력을 단일 지점에 모으고, unknown-subcommand 핸들러가 Commander 커맨드 트리를 single source of truth 로 사용하도록 한 설계가 깔끔합니다. 인라인으로 비차단 nit 1건(`extractNamespaceCommand` 의 value-taking 글로벌 옵션 skip-list 가 향후 drift 가능)을 남겼으나 현 시점 동작에는 영향 없습니다.

원본 이슈의 모든 수용 조건이 빠짐없이 반영되었고, 회귀 테스트 커버리지·스모크·빌드가 모두 통과하여 추가 핑퐁 없이 **Approve** 합니다. 🎉

---
_Generated by [Claude Code](https://claude.ai/code)_
